compiler/src/iree/compiler/Codegen/ExternalInterfaces/CPUEncodingExternalModels.cpp

```diff
@@ -725,11 +724,21 @@ struct CPUEncodingResolverMaterializerAttr final
     if (!linalgOp) {
       return nullptr;
     }
-
-    FailureOr<Operation *> newOp = lowerContractionOpWithEncoding(
-        b, linalgOp, convertedOperands,
-        cast<IREE::Encoding::LayoutMaterializerAttr>(layoutAttr));
-    return newOp.value_or(nullptr);
+    if (auto fillOp = dyn_cast<linalg::FillOp>(op)) {
```

Comment:
The implementations of `lowerOp` look practically the same, so maybe we extract this into a shared base class function?

Reply:
Yeah, I guess they might diverge in the future, but in this case both CPU and VMVX resolvers already have a common base class and implementations for different interface methods so I imo having them shared for now and duplicating when they start diverging would be best. However, I don't have the full context on how likely they will diverge in the future, so fine by me as well if you prefer to keep them separate.